scylla-cdc-replicator/src/replicator_consumer.rs

```diff
@@ -474,21 +482,41 @@ impl ReplicatorConsumer {
         if let Some(added_elements) = data.get_value(column_name) {
             let added_elements = added_elements.as_map().unwrap();
             for (key, val) in added_elements {
-                values_for_update[1] = Some(key);
-                values_for_update[2] = Some(val);
+                let mut inner_values_for_update = Vec::new();
+                values_for_update.clone_into(&mut inner_values_for_update);
+                let mut key_actual = None;
+                let key = match key {
+                    CqlValue::Timeuuid(u) => {
+                        key_actual.replace(CqlValue::Uuid((*u).into()));
+                        key_actual.as_ref()
+                    }
+                    _ => Some(key),
+                };
+                inner_values_for_update[1] = key;
+                inner_values_for_update[2] = Some(val);
```

Comment:
this hack is needed, at least for `scylla-cdc-rust v0.12.0` (haven't check for newer one), on update of non-frozen list, `replicator_consumer.rs` use scylla function `SCYLLA_TIMEUUID_LIST_INDEX` which need `Uuid`.

Reply:
@abastian would you like to validate it? Since I got all the changes from you it's reasonable. But if not I can hack into it.

Reply:
I'll test if previous code still valid on `v0.13.1`

Reply:
It's really confusing that the function `SCYLLA_TIMEUUID_LIST_INDEX` accepts a uuid instead of a timeuuid... But I understand the reason now, previous versions of the driver didn't validate types during serialization.

However, I still don't understand why it is necessary to clone the whole `values_for_update`. It looks like it should be possible to just overwrite `values_for_update[1]` and `values_for_update[2]`.

Reply:
after reverting to previous ones (this and below), still failed for `scylla_cdc_replicator` at following tests

- `replication_tests::tests::test_list_replace`
- `replication_tests::tests::test_list_update`

with error message :
```
---- replication_tests::tests::test_list_replace stdout ----
unique_name: test_rust_1721843863_2
unique_name: test_rust_1721843864_17
thread 'replication_tests::tests::test_list_replace' panicked at scylla-cdc-replicator/src/replication_tests.rs:779:52:
called `Result::unwrap()` on an `Err` value: Serializing values failed: SerializationError: Failed to serialize query arguments &[core::option::Option<&scylla_cql::frame::response::result::CqlValue>]: failed to serialize column uuid_idx(v): SerializationError: Failed to type check Rust type scylla_cql::frame::response::result::CqlValue against CQL type Uuid: expected one of the CQL types: [Timeuuid]
```

and,
```
---- replication_tests::tests::test_list_update stdout ----
unique_name: test_rust_1721843863_5
unique_name: test_rust_1721843864_20
thread 'replication_tests::tests::test_list_update' panicked at scylla-cdc-replicator/src/replication_tests.rs:737:52:
called `Result::unwrap()` on an `Err` value: Serializing values failed: SerializationError: Failed to serialize query arguments &[core::option::Option<&scylla_cql::frame::response::result::CqlValue>]: failed to serialize column uuid_idx(v): SerializationError: Failed to type check Rust type scylla_cql::frame::response::result::CqlValue against CQL type Uuid: expected one of the CQL types: [Timeuuid]
```

Reply:
ic... you mean why cloning `values_for_update` to `inner_values_for_update` ...
hmm.. this might be changed to cloning only when the key is `TimeUuid`, I'll see what can be done here tomorrow, I live at GMT+7 here... :)
btw, I think better solution is for `SCYLLA_TIMEUUID_LIST_INDEX` able to receive `TimeUuid`, since this is more inline with the function name ...

Reply:
> hmm.. this might be changed to cloning only when the key is `TimeUuid`, I'll see what can be done here tomorrow, I live at GMT+7 here... :)

Sure, no pressure.

> btw, I think better solution is for SCYLLA_TIMEUUID_LIST_INDEX able to receive TimeUuid, since this is more inline with the function name ...

I'll double check whether SCYLLA_TIMEUUID_LIST_INDEX accepts uuid or timeuuid with a different driver. I have a hunch this might be a bug in the rust driver.

Reply:
Thanks for the contribution, @abastian! You can ping me out here or in any social (danielhe4rt) if you need anything.